lib/paseto/local.rb

```diff
@@ -1,62 +1,69 @@
 module Paseto
-  class Local
-    HEADER = 'v2.local'
-    NONCE_BYTES = RbNaCl::AEAD::ChaCha20Poly1305IETF.nonce_bytes
+  module V2
+    module Local
+      HEADER = 'v2.local'
+      NONCE_BYTES = RbNaCl::AEAD::ChaCha20Poly1305IETF.nonce_bytes
 
-    NonceError = Class.new(Paseto::Error)
-    AuthError = Class.new(Paseto::Error)
-    BadMessageError = Class.new(Paseto::Error)
+      NonceError = Class.new(Paseto::Error)
 
-    def self.generate_aead_key
-      RbNaCl::Random.random_bytes(RbNaCl::AEAD::ChaCha20Poly1305IETF.key_bytes)
-    end
+      class Key
+        def self.generate
+          new(RbNaCl::Random.random_bytes(RbNaCl::AEAD::ChaCha20Poly1305IETF.key_bytes))
+        end
 
-    def self.generate_nonce
-      RbNaCl::Random.random_bytes(NONCE_BYTES)
-    end
+        def self.decode64(encoded_key)
+          new(Paseto.decode64(encoded_key))
+        end
 
-    def self.from_encode64_key(encoded_key, footer = nil)
-      new(Paseto.decode64(encoded_key), footer)
-    end
+        def initialize(key)
+          @key = key
+          @aead = RbNaCl::AEAD::ChaCha20Poly1305IETF.new(key)
+        end
 
-    def initialize(private_key, footer = nil)
-      @aead = RbNaCl::AEAD::ChaCha20Poly1305IETF.new(private_key)
-      @footer = footer
-    end
+        def encode64
+          Paseto.encode64(@key)
+        end
+
+        attr_reader :aead
+      end
 
-    def encrypt(message)
-      # Make a nonce: A single-use value never repeated under the same key
-      nonce = self.class.generate_nonce
+      def self.generate_nonce
```

Comment:
Stylistic comment: class methods should hang out together at the top of the class def